build/build.ps1

```diff
@@ -0,0 +1,246 @@
+$ErrorActionPreference = "Stop"
+
+function Print-Usage() {
+    Write-Host "Common settings:"
+    Write-Host "  -configuration <value>  Build configuration Debug, Release"
+    Write-Host "  -verbosity <value>      Msbuild verbosity (q[uiet], m[inimal], n[ormal], d[etailed], and diag[nostic])"
+    Write-Host "  -help                   Print help and exit"
+    Write-Host ""
+    Write-Host "Actions:"
+    Write-Host "  -restore                Restore dependencies"
+    Write-Host "  -build                  Build solution"
+    Write-Host "  -rebuild                Rebuild solution"
+    Write-Host "  -deploy                 Deploy built VSIXes"
+    Write-Host "  -test                   Run all unit tests in the solution"
+    Write-Host "  -sign                   Sign build outputs"
+    Write-Host "  -pack                   Package build outputs into NuGet packages and Willow components"
+    Write-Host ""
+    Write-Host "Advanced settings:"
+    Write-Host "  -solution <value>       Path to solution to build"
+    Write-Host "  -ci                     Set when running on CI server"
+    Write-Host "  -log                    Enable logging (by default on CI)"
+    Write-Host "  -prepareMachine         Prepare machine for CI run"
+    Write-Host "  -fullMSBuild            Test against the desktop version of MSBuild"
+    Write-Host ""
+    Write-Host "Command line arguments not listed above are passed through to MSBuild."
+    Write-Host "The above arguments can be shortened as much as to be unambiguous (e.g. -co for configuration, -t for test, etc.)."
+}
+
+function Create-Directory([string[]] $path) {
+  if (!(Test-Path -path $path)) {
+    New-Item -path $path -force -itemType "Directory" | Out-Null
+  }
+}
+
+function GetDotNetCliVersion {
+  [xml]$xml = Get-Content $DependenciesProps
+
+  foreach ($propertyGroup in $xml.Project.PropertyGroup) {
+    if (Get-Member -inputObject $propertyGroup -name dotnetCliVersion) {
+        return $propertyGroup.DotNetCliVersion
+    }
+  }
+
+  throw "Failed to locate the .NET CLI Version"
+}
+
+function GetVSWhereVersion {
+  [xml]$xml = Get-Content $DependenciesProps
+
+  foreach ($propertyGroup in $xml.Project.PropertyGroup) {
+    if (Get-Member -inputObject $propertyGroup -name VSWhereVersion)
+    {
+        return $propertyGroup.VSWhereVersion
+    }
+  }
+
+  throw "Failed to locate the VSWhere Version"
+}
+
+function InstallDotNetCli {
+  $dotnetCliVersion = GetDotNetCliVersion
+  $dotnetInstallVerbosity = ""
+
+  if (!$env:DOTNET_INSTALL_DIR) {
+    $env:DOTNET_INSTALL_DIR = Join-Path $RepoRoot "artifacts\.dotnet\$dotnetCliVersion"
+  }
+
+  $dotnetInstallScript = Join-Path $env:DOTNET_INSTALL_DIR "dotnet-install.ps1"
+
+  if (!(Test-Path $dotnetInstallScript)) {
+    Create-Directory $env:DOTNET_INSTALL_DIR
+    Invoke-WebRequest "https://raw.githubusercontent.com/dotnet/cli/master/scripts/obtain/dotnet-install.ps1" -UseBasicParsing -OutFile $dotnetInstallScript
```

Comment:
We should use this URL: https://dot.net/v1/dotnet-install.ps1

Reply:
will adjust. Is which script to use documented somewhere?

Reply:
This will always have the latest released script available and they are backwards compatible.